django/db/backends/mysql/base.py

```diff
@@ -360,6 +360,8 @@ def check_constraints(self, table_names=None):
                     )
 
     def is_usable(self):
+        if self.connection is None:
+            return False
```

Comment:
@cjerdonek I got your point. but if I follow you I'll end up with something that look like:

```python
    def is_usable(self):
        try:
            self.connection.ping()
        except Database.Error:
            return False
        except AttributeError:
            if connection is None:
                return False
            else:
                raise
        else:
            return True

```
Because I am not sure at all that we want to return `False` for every `AttributeError`. We want to explicitly handle the case where `connection` is `None`. Don't you think the first version is easier to read ?